File: app/src/main/java/org/stepic/droid/core/LocalLessonSessionManager.java
package org.stepic.droid.core;

import org.jetbrains.annotations.Nullable;
import org.stepic.droid.model.Attempt;
import org.stepic.droid.model.Submission;

import java.util.HashMap;
import java.util.Map;

import javax.inject.Inject;

public class LocalLessonSessionManager implements ILessonSessionManager {

    private final String TAG = "Lesson Session Manager";
    private Map<Long, Attempt> mStepIdToAttemptMap;
    private Map<Long, Submission> mStepIdToSubmissionMap;

    @Inject
    public LocalLessonSessionManager() {
        mStepIdToAttemptMap = new HashMap<>();
        mStepIdToSubmissionMap = new HashMap<>();

    }

    @Nullable
    @Override
    public Submission restoreSubmissionForStep(long stepId) {
        return mStepIdToSubmissionMap.get(stepId);
    }

    @Nullable
    @Override
    public Attempt restoreAttemptForStep(long stepId) {
        return mStepIdToAttemptMap.get(stepId);
    }

    @Override
    public void saveSession(long stepId, @Nullable Attempt attempt, @Nullable Submission submission) {
        if (attempt == null || submission == null) {
            return;
        }

        mStepIdToAttemptMap.put(stepId, attempt);
        mStepIdToSubmissionMap.put(stepId, submission);
    }

    @Override
    public void reset() {
        mStepIdToAttemptMap.clear();
        mStepIdToSubmissionMap.clear();
    }
}

File: app/src/main/java/org/stepic/droid/core/LocalLessonSessionManager.kt
package org.stepic.droid.core

import org.stepic.droid.model.Attempt
import org.stepic.droid.model.LessonSession
import org.stepic.droid.model.Submission
import java.util.*
import javax.inject.Inject

class LocalLessonSessionManager
@Inject constructor() : ILessonSessionManager {

    private val stepIdToLessonSession: MutableMap<Long, LessonSession>

    init {
        stepIdToLessonSession = HashMap<Long, LessonSession>()
    }

    override fun saveSession(stepId: Long, attempt: Attempt?, submission: Submission?, numberOfSubmissionOnFirstPage: Int) {
        if (attempt == null || submission == null) {
            return
        }

        stepIdToLessonSession.put(stepId, LessonSession(stepId, attempt, submission, numberOfSubmissionOnFirstPage))
    }

    override fun reset() {
        stepIdToLessonSession.clear()
    }

    override fun restoreLessonSession(stepId: Long): LessonSession? {
        return stepIdToLessonSession[stepId]
    }
}
